### ISSUE TITLE
Type.typeof(Int64.make(0, 0)) yields different results between different platforms

### ISSUE THREAD

**Issue by Atry** [HaxeFoundation/haxe#3096]
- `TInt` or `TDouble` for Java targets
- `TClass(haxe.Int64)` for Flash targets
- `TClass(haxe.Int64)` for Neko targets

**Comment by Simn** [comment]
This can (and should) be addressed now.

**Comment by ncannasse** [comment]
Reflection on abstracts is problematic.

We could add a new TInt64 result, but are we sure we can correctly deduce it on all platforms ? (no false positive, and no missed value) + this will require additional things in (Un)Serializer

**Comment by Simn** [comment]
I think the main point here is that Int64 can't be distinguished from Int and Float on these targets.

**Comment by waneck** [comment]
Should we add a new TInt64 result?

**Comment by ncannasse** [comment]
@waneck I think that's a bit early for that, same for supporting Int64 in serializer. Let's way a bit more until it becomes a first class citizen.

**Comment by Simn** [comment]
The problem right now is that people who used `Std.is(v, Int64)` do not have an alternative.

**Comment by ncannasse** [comment]
@Simn we could add a Int64.is for the time being

**Comment by ncannasse** [comment]
(which would be not exclusive, which means an Int could be Int64 if that's the way it's implemented on the platform)

**Comment by Simn** [comment]
What about serialization btw?

**Comment by ncannasse** [comment]
Same, I think we should wait for 3.3 to fully support Int64 in terms of reflection/serialization, if it needs to be (which i'm not yet sure)

**Comment by jonasmalacofilho** [comment]
First, I think the behavior in Java should be considered a bug, since `Type.typeof` will return TFloat for values that can't be represented at all with a double. Even if no more Int64 changes are pushed to other targets, fixing this in 3.2 would be awesome.

Second, a non exclusive `Int64.is` if fine. Or, how about adding `TInt64` to ValueType, as proposed on IRC by Simn?

Finally, the point of the recent changes hasn't been to improve the usage Int64? This suggests that it should be a first class citizen in Haxe, thus supported for reflection/serialization as soon as possible.

**Comment by Simn** [comment]
I can't see us addressing this in any clean way without adding TInt64.

**Comment by waneck** [comment]
Should we add a TInt64 type?

**Comment by Simn** [comment]
I think so.

**Comment by waneck** [comment]
I agree. I can do Java/C#'s part, but we'll need it to be implemented on the other targets as well

**Comment by Simn** [comment]
Start a branch then!
